src/components/FolderCreationModal.jsx

```diff
@@ -0,0 +1,237 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import { Base64 } from 'js-base64';
+import PropTypes from 'prop-types';
+import * as _ from 'lodash';
+import { Redirect } from 'react-router-dom';
+import update from 'immutability-helper';
+import Select from 'react-select';
+import FormField from './FormField';
+import FolderCard from './FolderCard';
+import LoadingButton from './LoadingButton';
+import elementStyles from '../styles/isomer-cms/Elements.module.scss';
+import contentStyles from '../styles/isomer-cms/pages/Content.module.scss';
+import adminStyles from '../styles/isomer-cms/pages/Admin.module.scss';
+import { validateCategoryName } from '../utils/validators';
+import { toast } from 'react-toastify';
+import Toast from './Toast';
+
+// axios settings
+axios.defaults.withCredentials = true
+
+const FolderCreationModal = ({
+  parentFolder,
+  existingSubfolders,
+  pagesData,
+  siteName,
+  setIsFolderCreationActive,
+}) => {
+  // Errors
+  const [errors, setErrors] = useState('')
+
+  const [folderName, setFolderName] = useState('');
+  const [title, setTitle] = useState('')
+  const [selectedFiles, setSelectedFiles] = useState(new Set())
+  //retrieve only when necessary, i.e. after user has chosen to move this page
+  // const [sha, setSha] = useState('')
+
+  const [isSelectingTitle, setIsSelectingTitle] = useState(true)
+  // Page redirection and modals
+  const [newPageUrl, setNewPageUrl] = useState('')
+  const [redirectToNewPage, setRedirectToNewPage] = useState(false)
+  const [sortedPagesData, setSortedPagesData] = useState(pagesData)
+
+  const sortOptions = [
+    ... parentFolder 
+      ? [{
+        value: 'collection',
+        label: 'Collection',
```

Comment:
Since we've agreed to move away from using the term `Collection`, can we use another term for the default order?

Reply:
Good point, fixed in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/f1a2924858d087d7fbdc0468fbf3b9ed30541bf6